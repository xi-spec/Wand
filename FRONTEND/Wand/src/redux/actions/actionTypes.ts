export default {
  LOAD_PRODUCT_TYPE: 'LOAD_PRODUCT_TYPE',
  LOAD_ALL_PRODUCTS: 'LOAD_ALL_PRODUCTS',
  LOAD_NEW_PRODUCTS: 'LOAD_NEW_PRODUCTS',
  LOAD_PRODUCTS_LIST: 'LOAD_PRODUCTS_LIST',
  LOAD_PRODUCT_DETAIL: 'LOAD_PRODUCT_DETAIL',
  ADD_PRODUCT_TO_CART: ' ADD_PRODUCT_TO_CART',
  INCREASE_QUANTITY_CART: 'INCREASE_QUANTITY_CART',
  DECREASE_QUANTITY_CART: 'DECREASE_QUANTITY_CART',
  DELETE_PRODUCT_CART: 'DELETE_PRODUCT_CART',
  DELETE_ALL_CART: 'DELETE_ALL_CART',
  UPDATE_STOCK_API: 'UPDATE_STOCK_API',
  SEARCH_PRODUCT: 'SEARCH_PRODUCT',
  ADD_PRODUCT_TO_FAVOURITE: 'ADD_PRODUCT_TO_FAVOURITE',
  REGISTER: 'REGISTER',
  REGISTER_ERROR: 'REGISTER_ERROR',
  LOGIN: 'LOGIN',
  LOGIN_ERROR: 'LOGIN_ERROR',
  CLEAN_ISEXIST_STATUS: 'CLEAN_STATUS',
  CLEAN_LOGIN_ERROR: 'CLEAN_LOGIN_ERROR',
  CLEAN_REGISTED: 'CLEAN_REGISTED',
  REMOVE_FAVOURITE: 'REMOVE_FAVOURITE',
  LOGOUT: 'LOGOUT',
  UPDATE_USER: 'UPDATE_USER',
  PAYMENT_SUCCSSFULY: 'PAYMENT_SUCCESSFULY',
  PDATE_STOCK_API: 'PDATE_STOCK_API'
}
;
