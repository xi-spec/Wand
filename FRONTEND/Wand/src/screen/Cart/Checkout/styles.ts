import { StyleSheet } from 'react-native';
import { COLORS, SIZE, FONTS } from '../../../constants/theme';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    margin: 10
  },
  shippingContainer: {
    marginTop: 20,
    marginHorizontal: 30
  },
  titleText: {
    fontFamily: FONTS.logoRegular,
    fontSize: SIZE.primary
  },
  edit: {
    fontFamily: FONTS.logoRegular,
    fontSize: SIZE.primary,
    color: COLORS.primary
  },
  text: {
    width: 250,
    fontSize: SIZE.second,
    fontFamily: FONTS.primary
  },
  shadowBox: {
    height: 150,
    justifyContent: 'center',
    paddingHorizontal: 20,
    shadowColor: 'grey',
    shadowOffset: {
      width: 0,
      height: 2
    },
    shadowOpacity: 0.15,
    shadowRadius: 3.84,

    elevation: 5,
    backgroundColor: 'white',
    borderRadius: 5
  },
  rowContainer: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    paddingVertical: 20
  },
  iconContainer: {
    flexDirection: 'row',
    marginVertical: 15
  },
  icon: {
    marginRight: 10
  },
  card: {
    width: 80,
    height: 50,
    borderRadius: 5,
    marginRight: 30
  },
  cardNumber: {
    width: 150,
    paddingTop: 20
  },
  checkoutContainer: {
    width: '100%',
    height: 80,
    position: 'absolute',
    bottom: 25,
    backgroundColor: 'white'
  },

  totalContainer: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    margin: 10
  },
  totalText: {
    fontSize: SIZE.primary,
    fontFamily: FONTS.primary
  },
  checkout: {
    width: '100%',
    height: 50,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: COLORS.second,
    borderRadius: 5
  },
  checkoutText: {
    color: 'white',
    fontFamily: FONTS.logoRegular,
    fontSize: SIZE.primary

  }
});

export default styles;
