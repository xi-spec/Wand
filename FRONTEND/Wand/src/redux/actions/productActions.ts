import actionTypes from './actionTypes';
import axios from 'axios';
import SERVER from '../../constants/server';
import { Dispatch } from 'redux';
import { Product, ShoppingCart } from '../../models/initiaState';

export function loadProductsType () {
  return async (dispatch : Dispatch) => {
    const { data } = await axios.get(SERVER.URL_TYPE);
    dispatch({
      type: actionTypes.LOAD_PRODUCT_TYPE,
      data
    });
  };
}

export function loadAllProducts () {
  return async (dispatch : Dispatch) => {
    const { data } = await axios.get(SERVER.URL_PRODUCT);
    dispatch({
      type: actionTypes.LOAD_ALL_PRODUCTS,
      data
    });
  };
}

export function loadProductsList (typeName:string) {
  return {
    type: actionTypes.LOAD_PRODUCTS_LIST,
    typeName
  };
}

export function loadProductDetail (productId:string) {
  return {
    type: actionTypes.LOAD_PRODUCT_DETAIL,
    productId
  };
}

export function addProductToCart (product:Product) {
  return {
    type: actionTypes.ADD_PRODUCT_TO_CART,
    product
  };
}

export function increaseQuantityCart (productId:string) {
  return {
    type: actionTypes.INCREASE_QUANTITY_CART,
    productId
  };
}

export function decreaseQuantityCart (productId:string) {
  return {
    type: actionTypes.DECREASE_QUANTITY_CART,
    productId
  };
}

export function deleteProductCart (productId:string, quantity:number) {
  return {
    type: actionTypes.DELETE_PRODUCT_CART,
    productId,
    quantity
  };
}

export function searchProducts (name:string) {
  return {
    type: actionTypes.SEARCH_PRODUCT,
    name
  };
}

export function updateStockApi (cart:ShoppingCart) {
  return (dispatch : Dispatch) => {
    cart.cartList.forEach(async (product) => {
      const { data } = await axios.put(`${SERVER.URL_PRODUCT}/${product._id}`, { stock: product.stock });
      dispatch({
        type: actionTypes.UPDATE_STOCK_API,
        data
      });
    });
  };
}
