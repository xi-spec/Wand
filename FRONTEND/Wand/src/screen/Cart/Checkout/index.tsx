import React from 'react';
import { Text, View, TouchableOpacity, Image } from 'react-native';
import styles from './styles';

import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import { updateStockApi } from '../../../redux/actions/productActions';
import { paymentSuccessfuly } from '../../../redux/actions/userActions';
import { ShoppingCart, User } from '../../../models/initiaState';
import { Icon } from 'react-native-elements';
import { IMAGES } from '../../../constants/images';

function Checkout ({ navigation, cart, user, actions }:
  {cart:ShoppingCart, actions:any, user:User, navigation:{navigate:Function}}) {
  const { cardNumber } = user.card || { undefined };

  function confirmAndPay (user, cart) {
    navigation.navigate('Payment successful');
    actions.updateStockApi(cart);
    actions.paymentSuccessfuly(user, cart);
  }

  return (
    <View style={styles.container}>
    <View style={styles.shippingContainer}>
      <View style={styles.rowContainer}>
        <Text style={styles.titleText}>Shipping information</Text>
        <TouchableOpacity
        testID='navigation'
        onPress={() => navigation.navigate('Address')}>
        <Text style={styles.edit}>edit</Text>
        </TouchableOpacity>
      </View>
        <View style={styles.shadowBox}>
      <View style={styles.iconContainer}>
          <Icon style={styles.icon} name='person-outline' type='ionicon' />
          <Text style={styles.text}>{user.name}</Text>
      </View>

      <View style={styles.iconContainer}>
          <Icon style={styles.icon}name='map-outline' type='ionicon' />
          {user.address
            ? <Text style={styles.text}>{`${user.address.street} ${user.address.city} ${user.address.postcode} ${user.address.country}`}</Text>
            : <Text style={styles.text}>Please enter your shipping address </Text>
           }
      </View>
        </View>
    </View>

    <View style={styles.shippingContainer}>
      <View style={styles.rowContainer}>
        <Text style={styles.titleText}>Payment Method</Text>
        <TouchableOpacity
        testID='navigation'
        onPress={() => navigation.navigate('Payment')}>
        <Text style={styles.edit}>edit</Text>
        </TouchableOpacity>
      </View>
        <View style={styles.shadowBox}>
      <View style={styles.iconContainer}>
        <Image style={styles.card}source={{ uri: IMAGES.card }}></Image>
          {cardNumber
            ? <Text style={styles.cardNumber }>{`**** **** **** ${cardNumber.slice(cardNumber.length - 4)}`}</Text>
            : <Text style={styles.cardNumber }>Please enter your card information</Text>
          }
      </View>
        </View>
    </View>

    <View style={styles.checkoutContainer}>
          <View style={styles.totalContainer}>
          <Text style={styles.totalText}>Total:</Text>
          <Text style={styles.totalText}>{`${cart.totalCost.toFixed(2)} euros`}</Text>
          </View>

          <TouchableOpacity
          disabled={!user.address || !user.card}
          testID='pay'
          onPress={() => confirmAndPay(user, cart)}
           style={styles.checkout}>
            <Text style={styles.checkoutText}>CONFIRM AND PAY</Text>
          </TouchableOpacity>
        </View>

    </View>

  );
}

function mapStateToProps ({ cart, user }) {
  return {
    cart,
    user
  };
}

function mapDispatchToProps (dispatch) {
  return {
    actions: bindActionCreators({
      paymentSuccessfuly,
      updateStockApi
    }, dispatch)
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(Checkout);
