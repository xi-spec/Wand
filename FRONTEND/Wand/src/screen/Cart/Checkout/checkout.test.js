import { render, fireEvent } from '@testing-library/react-native';
import React from 'react';
import ConfigureStore from 'redux-mock-store';
import * as productActions from '../../../redux/actions/productActions';
import * as actions from '../../../redux/actions/userActions';

import { Provider } from 'react-redux';
import Checkout from './index';

describe('Given a Checkout component', () => {
  const mockStore = ConfigureStore();
  const store = mockStore({
    cart: {
      cartList: [{
        _id: '1',
        images: { alls: ['url'] },
        cost: { value: 1, currency: 'euro' }
      }],
      totalItem: 0,
      totalCost: 1
    },
    user: {
      address: {},
      card: { cardNumber: '123456789012' }

    }
  });
  let navigation;

  beforeEach(() => {
    jest.spyOn(actions, 'paymentSuccessfuly').mockReturnValueOnce({ type: '' });
    jest.spyOn(productActions, 'updateStockApi').mockReturnValueOnce({ type: '' });

    navigation = {
      navigate: jest.fn()
    };
  });

  describe('When is invoked ', () => {
    test('Then is rendered correct with user address & card information', () => {
      const tree = render(<Provider store={store}><Checkout /></Provider>);

      expect(tree).toMatchSnapshot();
    });
    test('Then is rendered correct without user address & card information', () => {
      const store = mockStore({
        cart: {
          cartList: [{
            _id: '1',
            images: { alls: ['url'] },
            cost: { value: 1, currency: 'euro' }
          }],
          totalItem: 0,
          totalCost: 1
        },
        user: {

        }
      });
      const tree = render(<Provider store={store}><Checkout /></Provider>);

      expect(tree).toMatchSnapshot();
    });
  });

  describe('When navigation button is pressed', () => {
    test('Then navigation.navigate is invoked', () => {
      const { getAllByTestId } = render(
          <Provider store={store}><Checkout navigation={navigation} /></Provider>);

      const navigationArray = getAllByTestId('navigation');

      navigationArray.forEach(navigation => fireEvent.press(navigation));

      expect(navigation.navigate).toHaveBeenCalled();
    });
  });

  describe('When pay button is pressed', () => {
    test('Then is rendered correctly', () => {
      const tree = render(
          <Provider store={store}><Checkout navigation={navigation} /></Provider>);
      fireEvent.press(tree.getByTestId('pay'));

      expect(tree).toMatchSnapshot();
    });
  });
});
