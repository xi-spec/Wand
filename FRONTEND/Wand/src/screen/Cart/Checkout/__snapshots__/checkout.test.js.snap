// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a Checkout component When is invoked  Then is rendered correct with user address & card information 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "margin": 10,
    }
  }
>
  <View
    style={
      Object {
        "marginHorizontal": 30,
        "marginTop": 20,
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingVertical": 20,
        }
      }
    >
      <Text
        style={
          Object {
            "fontFamily": "RalewayRegular",
            "fontSize": 18,
          }
        }
      >
        Shipping information
      </Text>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID="navigation"
      >
        <Text
          style={
            Object {
              "color": "#2E3192",
              "fontFamily": "RalewayRegular",
              "fontSize": 18,
            }
          }
        >
          edit
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": 5,
          "elevation": 5,
          "height": 150,
          "justifyContent": "center",
          "paddingHorizontal": 20,
          "shadowColor": "grey",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.15,
          "shadowRadius": 3.84,
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "marginVertical": 15,
          }
        }
      >
        <View
          style={
            Object {
              "overflow": "hidden",
            }
          }
        >
          <View
            style={
              Object {
                "marginRight": 10,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "justifyContent": "center",
                }
              }
            >
              <Text />
            </View>
          </View>
        </View>
        <Text
          style={
            Object {
              "fontFamily": "MontserratRegular",
              "fontSize": 14,
              "width": 250,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "marginVertical": 15,
          }
        }
      >
        <View
          style={
            Object {
              "overflow": "hidden",
            }
          }
        >
          <View
            style={
              Object {
                "marginRight": 10,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "justifyContent": "center",
                }
              }
            >
              <Text />
            </View>
          </View>
        </View>
        <Text
          style={
            Object {
              "fontFamily": "MontserratRegular",
              "fontSize": 14,
              "width": 250,
            }
          }
        >
          undefined undefined undefined undefined
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "marginHorizontal": 30,
        "marginTop": 20,
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingVertical": 20,
        }
      }
    >
      <Text
        style={
          Object {
            "fontFamily": "RalewayRegular",
            "fontSize": 18,
          }
        }
      >
        Payment Method
      </Text>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID="navigation"
      >
        <Text
          style={
            Object {
              "color": "#2E3192",
              "fontFamily": "RalewayRegular",
              "fontSize": 18,
            }
          }
        >
          edit
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": 5,
          "elevation": 5,
          "height": 150,
          "justifyContent": "center",
          "paddingHorizontal": 20,
          "shadowColor": "grey",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.15,
          "shadowRadius": 3.84,
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "marginVertical": 15,
          }
        }
      >
        <Image
          source={
            Object {
              "uri": "https://logodownload.org/wp-content/uploads/2016/10/visa-logo-8.png",
            }
          }
          style={
            Object {
              "borderRadius": 5,
              "height": 50,
              "marginRight": 30,
              "width": 80,
            }
          }
        />
        <Text
          style={
            Object {
              "paddingTop": 20,
              "width": 150,
            }
          }
        >
          **** **** **** 9012
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "white",
        "bottom": 25,
        "height": 80,
        "position": "absolute",
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "margin": 10,
        }
      }
    >
      <Text
        style={
          Object {
            "fontFamily": "MontserratRegular",
            "fontSize": 18,
          }
        }
      >
        Total:
      </Text>
      <Text
        style={
          Object {
            "fontFamily": "MontserratRegular",
            "fontSize": 18,
          }
        }
      >
        1.00 euros
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#F3603F",
          "borderRadius": 5,
          "height": 50,
          "justifyContent": "center",
          "opacity": 1,
          "width": "100%",
        }
      }
      testID="pay"
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontFamily": "RalewayRegular",
            "fontSize": 18,
          }
        }
      >
        CONFIRM AND PAY
      </Text>
    </View>
  </View>
</View>
`;

exports[`Given a Checkout component When is invoked  Then is rendered correct without user address & card information 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "margin": 10,
    }
  }
>
  <View
    style={
      Object {
        "marginHorizontal": 30,
        "marginTop": 20,
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingVertical": 20,
        }
      }
    >
      <Text
        style={
          Object {
            "fontFamily": "RalewayRegular",
            "fontSize": 18,
          }
        }
      >
        Shipping information
      </Text>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID="navigation"
      >
        <Text
          style={
            Object {
              "color": "#2E3192",
              "fontFamily": "RalewayRegular",
              "fontSize": 18,
            }
          }
        >
          edit
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": 5,
          "elevation": 5,
          "height": 150,
          "justifyContent": "center",
          "paddingHorizontal": 20,
          "shadowColor": "grey",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.15,
          "shadowRadius": 3.84,
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "marginVertical": 15,
          }
        }
      >
        <View
          style={
            Object {
              "overflow": "hidden",
            }
          }
        >
          <View
            style={
              Object {
                "marginRight": 10,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "justifyContent": "center",
                }
              }
            >
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "color": "#242424",
                      "fontSize": 24,
                    },
                    Object {
                      "backgroundColor": "transparent",
                    },
                    Object {
                      "fontFamily": "ionicons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
                testID="iconIcon"
              >
                
              </Text>
            </View>
          </View>
        </View>
        <Text
          style={
            Object {
              "fontFamily": "MontserratRegular",
              "fontSize": 14,
              "width": 250,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "marginVertical": 15,
          }
        }
      >
        <View
          style={
            Object {
              "overflow": "hidden",
            }
          }
        >
          <View
            style={
              Object {
                "marginRight": 10,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "justifyContent": "center",
                }
              }
            >
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "color": "#242424",
                      "fontSize": 24,
                    },
                    Object {
                      "backgroundColor": "transparent",
                    },
                    Object {
                      "fontFamily": "ionicons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
                testID="iconIcon"
              >
                
              </Text>
            </View>
          </View>
        </View>
        <Text
          style={
            Object {
              "fontFamily": "MontserratRegular",
              "fontSize": 14,
              "width": 250,
            }
          }
        >
          Please enter your shipping address 
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "marginHorizontal": 30,
        "marginTop": 20,
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingVertical": 20,
        }
      }
    >
      <Text
        style={
          Object {
            "fontFamily": "RalewayRegular",
            "fontSize": 18,
          }
        }
      >
        Payment Method
      </Text>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID="navigation"
      >
        <Text
          style={
            Object {
              "color": "#2E3192",
              "fontFamily": "RalewayRegular",
              "fontSize": 18,
            }
          }
        >
          edit
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": 5,
          "elevation": 5,
          "height": 150,
          "justifyContent": "center",
          "paddingHorizontal": 20,
          "shadowColor": "grey",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.15,
          "shadowRadius": 3.84,
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "marginVertical": 15,
          }
        }
      >
        <Image
          source={
            Object {
              "uri": "https://logodownload.org/wp-content/uploads/2016/10/visa-logo-8.png",
            }
          }
          style={
            Object {
              "borderRadius": 5,
              "height": 50,
              "marginRight": 30,
              "width": 80,
            }
          }
        />
        <Text
          style={
            Object {
              "paddingTop": 20,
              "width": 150,
            }
          }
        >
          Please enter your card information
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "white",
        "bottom": 25,
        "height": 80,
        "position": "absolute",
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "margin": 10,
        }
      }
    >
      <Text
        style={
          Object {
            "fontFamily": "MontserratRegular",
            "fontSize": 18,
          }
        }
      >
        Total:
      </Text>
      <Text
        style={
          Object {
            "fontFamily": "MontserratRegular",
            "fontSize": 18,
          }
        }
      >
        1.00 euros
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#F3603F",
          "borderRadius": 5,
          "height": 50,
          "justifyContent": "center",
          "opacity": 1,
          "width": "100%",
        }
      }
      testID="pay"
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontFamily": "RalewayRegular",
            "fontSize": 18,
          }
        }
      >
        CONFIRM AND PAY
      </Text>
    </View>
  </View>
</View>
`;

exports[`Given a Checkout component When pay button is pressed Then is rendered correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "margin": 10,
    }
  }
>
  <View
    style={
      Object {
        "marginHorizontal": 30,
        "marginTop": 20,
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingVertical": 20,
        }
      }
    >
      <Text
        style={
          Object {
            "fontFamily": "RalewayRegular",
            "fontSize": 18,
          }
        }
      >
        Shipping information
      </Text>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID="navigation"
      >
        <Text
          style={
            Object {
              "color": "#2E3192",
              "fontFamily": "RalewayRegular",
              "fontSize": 18,
            }
          }
        >
          edit
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": 5,
          "elevation": 5,
          "height": 150,
          "justifyContent": "center",
          "paddingHorizontal": 20,
          "shadowColor": "grey",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.15,
          "shadowRadius": 3.84,
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "marginVertical": 15,
          }
        }
      >
        <View
          style={
            Object {
              "overflow": "hidden",
            }
          }
        >
          <View
            style={
              Object {
                "marginRight": 10,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "justifyContent": "center",
                }
              }
            >
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "color": "#242424",
                      "fontSize": 24,
                    },
                    Object {
                      "backgroundColor": "transparent",
                    },
                    Object {
                      "fontFamily": "ionicons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
                testID="iconIcon"
              >
                
              </Text>
            </View>
          </View>
        </View>
        <Text
          style={
            Object {
              "fontFamily": "MontserratRegular",
              "fontSize": 14,
              "width": 250,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "marginVertical": 15,
          }
        }
      >
        <View
          style={
            Object {
              "overflow": "hidden",
            }
          }
        >
          <View
            style={
              Object {
                "marginRight": 10,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "justifyContent": "center",
                }
              }
            >
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "color": "#242424",
                      "fontSize": 24,
                    },
                    Object {
                      "backgroundColor": "transparent",
                    },
                    Object {
                      "fontFamily": "ionicons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
                testID="iconIcon"
              >
                
              </Text>
            </View>
          </View>
        </View>
        <Text
          style={
            Object {
              "fontFamily": "MontserratRegular",
              "fontSize": 14,
              "width": 250,
            }
          }
        >
          undefined undefined undefined undefined
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "marginHorizontal": 30,
        "marginTop": 20,
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingVertical": 20,
        }
      }
    >
      <Text
        style={
          Object {
            "fontFamily": "RalewayRegular",
            "fontSize": 18,
          }
        }
      >
        Payment Method
      </Text>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID="navigation"
      >
        <Text
          style={
            Object {
              "color": "#2E3192",
              "fontFamily": "RalewayRegular",
              "fontSize": 18,
            }
          }
        >
          edit
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": 5,
          "elevation": 5,
          "height": 150,
          "justifyContent": "center",
          "paddingHorizontal": 20,
          "shadowColor": "grey",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.15,
          "shadowRadius": 3.84,
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "marginVertical": 15,
          }
        }
      >
        <Image
          source={
            Object {
              "uri": "https://logodownload.org/wp-content/uploads/2016/10/visa-logo-8.png",
            }
          }
          style={
            Object {
              "borderRadius": 5,
              "height": 50,
              "marginRight": 30,
              "width": 80,
            }
          }
        />
        <Text
          style={
            Object {
              "paddingTop": 20,
              "width": 150,
            }
          }
        >
          **** **** **** 9012
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "white",
        "bottom": 25,
        "height": 80,
        "position": "absolute",
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "margin": 10,
        }
      }
    >
      <Text
        style={
          Object {
            "fontFamily": "MontserratRegular",
            "fontSize": 18,
          }
        }
      >
        Total:
      </Text>
      <Text
        style={
          Object {
            "fontFamily": "MontserratRegular",
            "fontSize": 18,
          }
        }
      >
        1.00 euros
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#F3603F",
          "borderRadius": 5,
          "height": 50,
          "justifyContent": "center",
          "opacity": 1,
          "width": "100%",
        }
      }
      testID="pay"
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontFamily": "RalewayRegular",
            "fontSize": 18,
          }
        }
      >
        CONFIRM AND PAY
      </Text>
    </View>
  </View>
</View>
`;
